scylla/src/transport/connection.rs

```diff
@@ -1721,6 +1728,112 @@ mod tests {
         assert!(insert_res1.is_empty());
     }
 
+    #[tokio::test]
+    #[cfg(not(scylla_cloud_tests))]
+    async fn test_coalescing() {
+        // It's difficult to write a reliable test that checks whether coalescing
+        // works like intended or not. Instead, this is a smoke test which is supposed
+        // to trigger the coalescing logic and check that everything works fine
+        // no matter whether coalescing is enabled or not.
+
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+        let addr: SocketAddr = resolve_hostname(&uri).await;
+        let ks = unique_keyspace_name();
+
+        {
+            // Preparation phase
+            let session = SessionBuilder::new()
+                .known_node_addr(addr)
+                .build()
+                .await
+                .unwrap();
+            session.query(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'SimpleStrategy', 'replication_factor' : 1}}", ks.clone()), &[]).await.unwrap();
+            session.use_keyspace(ks.clone(), false).await.unwrap();
+            session
+                .query(
+                    "CREATE TABLE IF NOT EXISTS t (p int primary key, v blob)",
+                    &[],
+                )
+                .await
+                .unwrap();
+        }
+
+        let subtest = |enable_coalescing: bool, ks: String| async move {
+            let (connection, _) = super::open_connection(
+                UntranslatedEndpoint::ContactPoint(ContactPoint {
+                    address: addr,
+                    datacenter: None,
+                }),
+                None,
+                ConnectionConfig {
+                    enable_write_coalescing: enable_coalescing,
+                    ..ConnectionConfig::default()
+                },
+            )
+            .await
+            .unwrap();
+            let connection = Arc::new(connection);
+
+            connection
+                .use_keyspace(&super::VerifiedKeyspaceName::new(ks, false).unwrap())
+                .await
+                .unwrap();
+
+            connection
+                .query(&"TRUNCATE t".into(), (), None)
+                .await
+                .unwrap();
+
+            let mut futs = Vec::new();
+
+            for batch_size in 0..10 {
+                // Each future should issue more and more queries in the first poll
+                let base = (batch_size - 1) * batch_size / 2;
```

Comment:
What does this formula mean?

Reply:
`n * (n - 1) / 2` = `1 + 2 + .. n - 1`

I'll extract it to a helper function and name it.